handlers/sourcetv.js

```diff
@@ -39,18 +36,29 @@ Dota2.Dota2Client.prototype.requestSourceTVGames = function(filter_options, call
 
 var handlers = Dota2.Dota2Client.prototype._handlers;
 
-var onSourceTVGamesResponse = function onSourceTVGamesResponse(message, callback) {
-    callback = callback || null;
+var onSourceTVGamesResponse = function onSourceTVGamesResponse(message) {
+    var sourceTVGamesResponse = Dota2.schema.CMsgGCToClientFindTopSourceTVGamesResponse.decode(message);
 
-    var sourceTVGamesResponse = Dota2.schema.CMsgSourceTVGamesResponse.decode(message);
-
-    if (typeof sourceTVGamesResponse.games !== "undefined" && sourceTVGamesResponse.games.length > 0) {
+    if (sourceTVGamesResponse.start_game !== null) {
         if (this.debug) util.log("Received SourceTV games data");
-        this.emit("sourceTVGamesData", sourceTVGamesResponse.num_total_games, sourceTVGamesResponse.games);
-        if (callback) callback(sourceTVGamesResponse);
+        this.emit("sourceTVGamesData", sourceTVGamesResponse);
     } else {
-        if (this.debug) util.log("Received a bad SourceTV games response");
-        if (callback) callback(sourceTVGamesResponse.result, sourceTVGamesResponse);
+        if (this.debug) util.log("Received a bad new SourceTV games response");
+        this.emit("sourceTVGamesData", sourceTVGamesResponse);
```

Comment:
I'm not convinced of this way of handling an error response. It's a bit cumbersome to have to ask the user to check for any of multiple possible key-value pairs to see if they are null. As far as I know, we usually either don't emit an event in case of error, or if we emit one, we give it a null argument. This makes it easier for the user, as he can simply check if that which he receives is null.